junit/src/main/java/org/kaazing/k3po/junit/rules/ScriptRunner.java

```diff
@@ -137,12 +139,19 @@ public ScriptPair call() throws Exception {
                     case FINISHED:
                         FinishedEvent finished = (FinishedEvent) event;
                         // note: observed script is possibly incomplete
-                        String observedScript = finished.getScript();
-                        return new ScriptPair(expectedScript, observedScript);
+                        observedScript = finished.getScript();
+                        
+                        // we keep the result and read any other event is still in the buffer. We will return after the first timeout
+//                        return new ScriptPair(expectedScript, observedScript);
+                        finishedReceived = true;
+                        break;
                     default:
                         throw new IllegalArgumentException("Unrecognized event kind: " + event.getKind());
                     }
                 } catch (SocketTimeoutException e) {
+                    if (finishedReceived)
```

Comment:
I have found a solution to stop sending to the client anything except the DISPOSED message after a FINISHED was sent. But this functionality means that we will have to change the specs also.

I did not commit the changes, as I'm not sure this is the best direction. There is also the option to have a loop in the junit side to read and ignore all events after FINISHED up to DISPOSED. This will not require any spec change.